spec/rester/client_spec.rb

```diff
@@ -252,47 +257,105 @@ def let_retry_period_pass
         let(:error_threshold) { 3 }
         let(:retry_period) { 0.001 }
 
-        before { error_threshold.times { error_request } }
+        before {
+          # 3 error request logs + 1 success req log + 1 success response log
+          expect(logger).to receive(:info).at_most(error_threshold + 2).times
+          error_threshold.times { error_request }
+        }
         after { let_retry_period_pass; success_request }
 
         it 'should log that circuit is now closed' do
-          expect(logger).to receive(:info).with('circuit closed').once
+          expect(logger).to receive(:info).with(
+            "Correlation-ID=#{Rester.correlation_id}: circuit closed for DummyService"
+          ).once
         end
       end # with error_threshold reached
+
+      context 'before first real request' do
+        before { success_request }
+
+        it 'should default producer name to to the service after first ping' do
+          expect(client.send(:_producer_name)).to eq "DummyService"
+        end
+      end # without producer name set
+
+      context 'with producer name set' do
+        before { success_request }
+
+        it 'should default producer name to "Producer"' do
+          expect(client.send(:_producer_name)).to eq "DummyService"
+        end
+      end # with producer name set
+
+      context 'with successful request' do
+        let(:logger) { double('logger') }
+        after { success_request }
+
+        it 'should log the correct messages' do
+          expect(logger).to receive(:info).with(
+            "Correlation-ID=#{Rester.correlation_id}: [Dummy] -> DummyService - GET /v1/ping"
+          ).once
+          expect(logger).to receive(:info).with(
+            "Correlation-ID=#{Rester.correlation_id}: [Dummy] <- DummyService - GET /v1/ping 200"
+          ).once
+        end
+      end # with successful request
+
+      context 'with client request_handler middleware simulation' do
+        let(:logger) { double('logger') }
+        before {
+          Rester.begin_request
+          Rester.correlation_id = SecureRandom.uuid
+        }
+        after {
```

Comment:
Does this context test any additional logic beyond what the `with successful request` context tests?  It doesn't look like it.  If not, then I think we should remove these tests. We only want to test the logic specific to the `#request` method.